documentation/spike/alert_emaails_spike.md

```diff
@@ -0,0 +1,84 @@
+
+## Background
+We currently send a huge number job alerts and we want to review this feature in anticipation of an increase once we expand into FE.
+
+We have no visibility on the number of job alert emails that are actually opened.
+
+## Desired outcomes
+Have a stable alert emails jobs memory and comptutationally wise even after the increased number of live vacancies.
+
+## Proposed suggestions
+After reviewing different alternatives to the issue at hand, the following set of suggestions seems to best fit TV.
+They could be applied sequentially without having to commit to a massive task at once.
+
+### Add alert email jobs monitoring
+This is to be able to observe the health of the service. We should monitor:
+- time taken to complete whole job
+- memory used to complete whole job
+- list the N slowest subscriptions to process
+- setup alerting when certain metric's threshold are reached
+
+### Pre-computed subscription.vacancies_matching
+Split subscription.vacancies_matching form email sending to have a predictable behaviour for the email sending even at peak times.
+
+The output of the pre-computation job should be optimized for retrieval so that the subsequent email sending job is as fast as possible.
+
+### Email content optimisation
+At the moment we could be loading up to 500 matching vacancies to send 1 email. This high number of vacancies matching could be reduced to 10-20 matches and the remaining matches could be accessed by adding a link to vacancy search page with the subscription.search_criterias filled in.
+
+### Caching
+The subscription.vacancies_matching pre-computation job output could be cached so that accessing the vacancy search page for a subscription.search_criteria does not impact the service when user following the link from their job alert email.
+Access on the link from an old email should be fine as long as the subscription is live (ie not soft/hard deleted). When the subscription has expired the service would perform fresh vacancy search operation.
+
+### Subscription retention policy
+We could soft delete any subscription that did not get its job alert email opened in that the last X time period.
+This would has 2 main benefits:
+- reducing the load for pre-computation and send email jobs
+- provided analytics to email opening rate for business stakeholders.
+Hard deletion is already in place.
+
+### Query optimisation
+Query optimisation of various degrees of difficulty could be further investgated to improve performance, here are some suggestions to that end:
+1. subscription.search_criteria grouping
+grouped by similar / identical subscription.search_criteria; the vacancies_matching ouput could be cached and the pre-computation skipped altogether [effort easy]
+
+2. simplify polygons for faster search results [effort medium]
+Change location search to use a center point with radius instead of polygon search.
+IE we take the center of the city, borough, region instead of the details entity borders.
```

Comment:
This isn't really feasible. It was done like this for a reason, changing the behaviour probably won't really help much. This was already optimised last year by reducing the number of points in the polygons